server/src/main/java/org/opensearch/action/search/SearchRequestOperationsListener.java

```diff
@@ -22,6 +22,16 @@
 @InternalApi
 public abstract class SearchRequestOperationsListener {
     private volatile boolean enabled;
+    public static final SearchRequestOperationsListener NOOP = new SearchRequestOperationsListener(false) {
```

Comment:
This is only needed for tracing, please move it to TraceableSearchRequestOperationsListener as a `private static  final` member

Reply:
Actually we can call protected `SearchRequestOperationsListener()` constructor from `TraceableSearchRequestOperationsListener` because it is a child class. 
However doesn't it make sense to leave at the parent level in case future listeners need a Noop implementation. Why create individual ones for each child class?